From 2f775a37d4e4cb432240b26b08525aff32f746f3 Mon Sep 17 00:00:00 2001
From: sin-ack <sin-ack@protonmail.com>
Date: Tue, 27 Oct 2026 11:29:13 -0700
Subject: [PATCH] build: Add a step for only building terminfo

Avoids building the rest of Ghostty which takes quite a while.
---
 src/build/GhosttyResources.zig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/build/GhosttyResources.zig b/src/build/GhosttyResources.zig
index e90cba9..3749cb3 100644
--- a/src/build/GhosttyResources.zig
+++ b/src/build/GhosttyResources.zig
@@ -92,6 +92,11 @@ pub fn init(b: *std.Build, cfg: *const Config, deps: *const SharedDeps) !Ghostty
                 .install_subdir = terminfo_share_dir,
             });
             try steps.append(b.allocator, &install_step.step);
+
+            // This step is to generate the terminfo data without building the rest of
+            // Ghostty which takes a long time.
+            const terminfo_step = b.step("terminfo", "Install the terminfo database");
+            terminfo_step.dependOn(&install_step.step);
         }
     }
 
